### PR TITLE
fix(sdk): fix a bug causing `_wandb` key being deleted from a run's config

### PR DESCRIPTION
**Pull request by fmamberti-wandb**
Description
-----------
- Fixes [WB-18096](https://wandb.atlassian.net/browse/WB-18096)

What does the PR do?

This fixes a bug causing the `_wandb` key being removed from the run.config if the Run was updated via the API.

- [X] I updated CHANGELOG.md, or it's not applicable

Testing
-------
How was this PR tested?

Updating the Config of a Run via API and confirming that the `_wandb` was not removed with the following code:

```python
import wandb
import time

entity = "fmamberti-wandb"
project = "debug-config"
id = "0iqlucv4"

api = wandb.Api()
run = api.run(f"{entity}/{project}/{id}")
wandb_key = run.rawconfig["_wandb"]
print(f"wandb key before update is: \n{wandb_key}")
run.config["model"] = "another_model_2"
run.update()

time.sleep(5)
api = wandb.Api()

updated_run = api.run(f"{entity}/debug-config/{id}")

assert updated_run.rawconfig["_wandb"] == wandb_key

assert updated_run.config["model"] == "another_model_2"
```

[WB-18096]: https://wandb.atlassian.net/browse/WB-18096?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by dmitryduev — CHANGES_REQUESTED** [review]
thanks @fmamberti-wandb! please fix the CI errors.

**Comment by fmamberti-wandb** [comment]
@dmitryduev I've now fixed the CI errors. Let me know if there is anything else needed to merge this.

File: tests/system_tests/test_core/test_public_api.py [hunk 1]
```diff
@@ -321,17 +321,21 @@ def test_run_create(user, relay_server):
 
 
 def test_run_update(user, relay_server):
-    seed_run = Api().create_run()
+    seed_run = wandb.init(config={"foo": "not_bar"})
```
Comment (dmitryduev) [inline tests/system_tests/test_core/test_public_api.py:324]: We prefer the wandb_init fixture, could you try this and see if it works?

```suggestion
def test_run_update(wandb_init, relay_server):
    seed_run = wandb_init(config={"foo": "not_bar"})
```

**Review by dmitryduev — APPROVED** [review]
one nit, but otherwise looks great!

File: tests/system_tests/test_core/test_public_api.py (hunk 1 above)
Comment (fmamberti-wandb) [reply tests/system_tests/test_core/test_public_api.py]: Done! thanks for the suggestion.

**Pull request merged**